system/Database/MigrationRunner.php

```diff
@@ -839,7 +839,8 @@ protected function migrate($direction, $migration): bool
             throw new RuntimeException($message);
         }
 
-        $instance = new $class();
+        /** @var Migration $instance */
+        $instance = new $class(Database::forge($this->db));
```

Comment:
This is not good.
The Migration class uses the DB connection in the DBForge if you pass it.
But it must use the `$this->DBGroup` if it has the property.

Reply:
Changed in https://github.com/codeigniter4/CodeIgniter4/pull/8221/commits/fa68681a7da63acc35417d420abd3149e9dc3352